src/Symfony/Bundle/DependencyInjection/Compiler/DeprecateLegacyIriConverterPass.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Symfony\Bundle\DependencyInjection\Compiler;
+
+use ApiPlatform\Core\Api\IriConverterInterface;
+use Symfony\Component\Config\Definition\BaseNode;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * @internal calls `setDeprecated` method with valid arguments depending on which version of symfony/dependency-injection is used
+ */
+final class DeprecateLegacyIriConverterPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        if ($container->hasAlias(IriConverterInterface::class)) {
+            $container
+                ->getAlias(IriConverterInterface::class)
+                ->setDeprecated(...$this->buildDeprecationArgs('Using "%alias_id%" is deprecated since API Platform 2.7. Use "ApiPlatform\Api\IriConverterInterface" instead.'));
+        }
+
+        if ($container->hasDefinition(IriConverterInterface::class)) {
+            $container
+                ->getDefinition(IriConverterInterface::class)
+                ->setDeprecated(...$this->buildDeprecationArgs('Using "%service_id%" is deprecated since API Platform 2.7. Use "ApiPlatform\Api\IriConverterInterface" instead.'));
+        }
+
+        if ($container->hasDefinition('api_platform.iri_converter.legacy')) {
+            $container
+                ->getDefinition('api_platform.iri_converter.legacy')
+                ->setDeprecated(...$this->buildDeprecationArgs('Using "%service_id%" is deprecated since API Platform 2.7. Use "ApiPlatform\Api\IriConverterInterface" instead.'));
+        }
```

Comment:
I would not deprecate this service, IMO its here to cover BC in some cases you want to use it.

Reply:
It's weird to introduce a deprecated service no? But indeed this service only helps covering a bc layer so why not! 

> Also by writing this table I'm realizing that ApiPlatform\Core\Api\IriConverterInterface is removed in 2.7 BC OFF, while the service api_platform.iri_converter.legacy still exist for the BC reasons you mentioned, shouldn't we also maintain the alias ApiPlatform\Core\Api\IriConverterInterface to api_platform.iri_converter.legacy in 2.7 BC OFF ?

definitely!